### DISCUSSION TITLE
Given a specific coordinate and a specific object, calculate times of conjunction for a given time span

### DISCUSSION THREAD

**Discussion by cnighswonger** [skyfielders/python-skyfield#671] [Q&A]
What I want to do is to specify an object, a coordinate, and a time span and be given back a list of conjunctions within some margin of error of the specified coordinate.

Is there any such high level functionality in Skyfield? Or will I need to dig in?

TIA

**Comment by brandon-rhodes** [comment]
That's a great question — the almanac module already has a search routine for conjunctions of planets with the Sun:

https://rhodesmill.org/skyfield/almanac.html#opposition-and-conjunction

It might be a simple adjustment to have a routine that does conjunctions between objects. Then you could give it a planet and, say, a Star object whose RA and Dec were at your coordinate of interest. My guess is that you'll want to dig in and try adapting the Sun-planet routine to look for conjunctions instead with a body the user passes in. Here's the code of the current routine:

https://github.com/skyfielders/python-skyfield/blob/2b3e23d065b1610c5eceb5110c2c272255990913/skyfield/almanac.py#L168-L189

See if you can get it working for something besides the Sun, I'll be interested to hear your results!

What kinds of conjunction are you searching for? Knowing a bit more about the problem would help us advise you on the approach.
